analysis/blacklist/blacklist.go

```diff
@@ -0,0 +1,223 @@
+package blacklist
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/ocmdev/mgosec"
+	bl "github.com/ocmdev/rita-bl"
+	blDB "github.com/ocmdev/rita-bl/database"
+	"github.com/ocmdev/rita-bl/list"
+	"github.com/ocmdev/rita-bl/sources/lists"
+	"github.com/ocmdev/rita-bl/sources/rpc"
+	"github.com/ocmdev/rita/config"
+	"github.com/ocmdev/rita/database"
+	log "github.com/sirupsen/logrus"
+	mgo "gopkg.in/mgo.v2"
+)
+
+type resultsChan chan map[string][]blDB.BlacklistResult
```

Comment:
We chunk up the checks against RITA-BL. RITA-BL returns maps of (hosts or ip addresses or urls) to their results. The channel is used for communication.